tests/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+
+# Build vector test
+add_executable(testVector testVector.cpp LinAlg/vectorTestsPar.cpp)
+target_link_libraries(testVector PRIVATE hiop ${HIOP_MATH_LIBRARIES})
+
+# Build matrix test
+add_executable(testMatrix testMatrix.cpp LinAlg/matrixTestsDense.cpp)
+
+target_link_libraries(testMatrix
+    PRIVATE
+        hiop_warnings
```

Comment:
@pelesh : target link with `hiop_math` instead of `hiop_warnings` and `hiop_options`. Otherwise the options specific to dev mode build will be enabled when they should not be

Reply:
I think this was my omission after a big rebase. By the way, don't we need to link to `hiop` (the whole thing) instead of `hiop_math`? The latter is just an interface to external math libraries.